backends/ebpf/psa/README.md

```diff
@@ -164,6 +164,103 @@ a new element to the outer map (indexed by clone session or multicast group iden
 or `multicast_group` in a PSA program) and initialize an inner map. To add a new clone session/multicast group member,
 a con1trol plane must add new element to the inner map.
 
+## P4 match kinds
+
+The PSA-eBPF compiler currently supports the following P4 match kinds: `exact`, `lpm`, `ternary`.
+
+### exact
+
+An `exact` table is implemented using the BPF hash map. A P4 table is considered an `exact` table if all its match fields are defined as `exact`.
+Then, the PSA-eBPF compiler generates a BPF hash map instance for each P4 table instance. The hash map key as a concatenation of P4 match fields translated to eBPF representation.
+Each `apply()` operation is translated into a lookup to the BPF hash map. The value is used to determine an action and its parameters. 
+
+### lpm
+
+An `lpm` table is implemented using the BPF `LPM_TRIE` map. A P4 table is considered an `lpm` table if it contains a single `lpm` field and no `ternary` fields.
+The PSA-eBPF compiler generates a BPF `LPM_TRIE` map instance for each P4 table instance. The hash map key as a concatenation of P4 match fields translated to eBPF representation.
+Moreover, the PSA-eBPF compiler shuffles the match fields and places the `lpm` field in the last position. Each `apply()` operation is translated into a lookup to the `LPM_TRIE` map.
+A control plane should populate the `LPM_TRIE` map with entries composed of a value and prefix. 
+
+### ternary
+
+There is no built-in BPF map for ternary (wildcard) matching. Hence, the PSA-eBPF compiler leverages the Tuple Space Search (TSS) algorithm for ternary matching (refer to the [research paper](https://dl.acm.org/doi/10.1145/316194.316216) to learn more about the TSS algorithm). 
+A `ternary` table is implemented using a combination of hash and array BPF maps that realizes the TSS algorithm. A P4 table is considered a `ternary` table if it contains at least one `ternary` field (exact and lpm fields are converted to ternary fields with an appropriate mask). 
+
+The PSA-eBPF compiler generates 2 BPF maps for each ternary table instance (+ the default action map):
+- the `<TBL-NAME>_prefixes` map is a BPF hash map that stores all unique ternary masks. The ternary masks are created based on the runtime table entries that are installed by a user.
+- the `<TBL-NAME>_tuples_map` map is a BPF array map of maps that stores all "tuples". A single tuple is a BPF hash map that stores all flow rules with the same ternary mask. 
+
+Note that the `psabpf-ctl table add` CLI command greatly simplifies the process of adding/removing flow rules to ternary tables. 
```

Comment:
I assume that this provides the illusion of a ternary table to the control-plane. This will be a consideration if you do p4Runtime too.

Reply:
Basically, `psabpf` provides not only a CLI tool, but also a C library that we plan to use, when it comes to the P4Runtime integration. So, the psabpf C API hides the underlying complexity.